lib/rucio/core/did.py

```diff
@@ -681,6 +683,12 @@ def delete_dids(dids, account, expire_rules=False, session=None):
             rowcount = session.query(models.CollectionReplica).filter(or_(*collection_replica_clause)).\
                 delete(synchronize_session=False)
 
+    # Remove generic did metadata
+    if metadata_to_delete:
+        with record_timer_block('undertaker.did_meta'):
+            rowcount = session.query(models.DidMeta).filter(or_(*metadata_to_delete)).\
```

Comment:
How does this work on Oracle < 12? Because in that case (E.g. like for ATLAS) the Table doesn't exist at all, but the core code will still try to execute this.

Reply:
If there are no metadata to delete, this code is not executed.

The question is what happens when
`metadata_to_delete.append(and_(models.DidMeta.scope == did['scope'], models.DidMeta.name == did['name']))`
is called on Oracle < 12.

Just to be safe, I could add both statements in a not oracle < 12 if statement.

After I am done with the Unification of the metadata checks like this will be managed there.

Reply:
Yes, because `metadata_to_delete` would be filled no matter what. And then the delete code bellow would also be called. I would add it to the append statement too just to be safe.

Reply:
Yes, you a right.
I will add the check to both statements for both pull requests in a few minutes.